JetAnalyzers/python/addAlgorithm.py

```diff
@@ -531,35 +555,38 @@ def addAlgorithm(process, alg_size_type_corr, Defaults, reco, doProducer):
     elif type == 'Calo':
         jra.srcRho = cms.InputTag("fixedGridRhoFastjetAllCalo")
     elif type == 'PFchs':
-        process.kt6PFchsJetsRhos = kt6PFJets.clone(src = 'pfNoPileUpJME',
+        process.pfCHS = cms.EDFilter("CandPtrSelector", src = cms.InputTag("packedPFCandidates"), cut = cms.string("fromPV"))
+        process.kt6PFchsJetsRhos = kt6PFJets.clone(src = 'pfCHS',
                                                    doFastJetNonUniform = cms.bool(True),
                                                    puCenters = cms.vdouble(-5,-4,-3,-2,-1,0,1,2,3,4,5), 
                                                    puWidth = cms.double(.8),
                                                    nExclude = cms.uint32(2))
-        sequence = cms.Sequence(process.kt6PFchsJetsRhos * sequence)
+        sequence = cms.Sequence(process.pfCHS * process.kt6PFchsJetsRhos * sequence)
         jra.srcRhos = cms.InputTag("kt6PFchsJetsRhos", "rhos")
         jra.srcRho = cms.InputTag("fixedGridRhoFastjetAll")
-        jra.srcPFCandidates = cms.InputTag('pfNoPileUpJME')
+        jra.srcPFCandidates = cms.InputTag('pfCHS')
     elif type == 'PFHLT':
-        jra.srcRho = ak4PFL1Fastjet.srcRho #added 02/15/2012
+        jra.srcRho = ak4PFL1Fastjet.srcRho 
         jra.srcRhoHLT = ak5PFHLTL1Fastjet.srcRho
     elif type == 'PFchsHLT':
-        jra.srcRho = ak4PFchsL1Fastjet.srcRho #added 02/15/2012
+        jra.srcRho = ak4PFchsL1Fastjet.srcRho 
         jra.srcRhoHLT = ak5PFchsHLTL1Fastjet.srcRho
     elif type == 'PF':
+        process.particleFlow = cms.EDFilter("CandPtrSelector", src = cms.InputTag("packedPFCandidates"), cut = cms.string(""))
```

Comment:
Why are you recreating the particleFlow collection without any cuts? Why not just use the packedPFCandidates directly as inputs on line 583? Same comment for lines 585-589.